server/schedulers/balance_plan.go

```diff
@@ -80,3 +105,54 @@ func (p *balanceSchedulerPlan) Clone(opts ...plan.Option) plan.Plan {
 	}
 	return plan
 }
+
+func BalancePlanSummary(plans []plan.Plan) (string, error) {
+	secondGroup := make(map[plan.Status]uint64)
+	var firstGroup map[uint64]map[plan.Status]int
+	maxStep := -1
+	for _, pi := range plans {
+		p, ok := pi.(*balanceSchedulerPlan)
+		if !ok {
+			return "", errs.ErrDiagnoseLoadPlanError
+		}
+		step := p.GetStep()
+		// we don't consider the situation for verification step
+		if step > 2 {
+			continue
+		}
+		if step > maxStep {
+			firstGroup = make(map[uint64]map[plan.Status]int)
+			maxStep = p.GetStep()
+		} else if step < maxStep {
+			continue
+		}
+		var store uint64
+		if step == 1 {
```

Comment:
Will it return early?

Reply:
it will not.

Reply:
Please add some comments to make it clear.